internal/server/stat_series.go

```diff
@@ -288,91 +288,88 @@ func (s *Server) GetStats(ctx context.Context, in *pb.GetStatsRequest) (
 }
 
-// GetStat implements API for Mixer.GetStat.
-// Endpoint: /stat/series/rich
-func (s *Server) GetStat(ctx context.Context, in *pb.GetStatRequest) (
-	*pb.GetStatResponse, error) {
-	placeDcids := in.GetPlaces()
-	statVarDcid := in.GetStatVar()
-	if len(placeDcids) == 0 {
-		return nil, status.Errorf(codes.InvalidArgument, "Missing required argument: place")
-	}
-	if statVarDcid == "" {
-		return nil, status.Errorf(codes.InvalidArgument, "Missing required argument: stat_var")
+// GetStatSetSeries implements API for Mixer.GetStatSetSeries.
+// Endpoint: /v1/stat/set/series
+func (s *Server) GetStatSetSeries(ctx context.Context, in *pb.GetStatSetSeriesRequest) (
+	*pb.GetStatSetSeriesResponse, error) {
+	places := in.GetPlaces()
+	statVars := in.GetStatVars()
+	if len(places) == 0 {
+		return nil, status.Errorf(
+			codes.InvalidArgument, "Missing required argument: places")
 	}
-	filterProp := &ObsProp{
-		Mmethod: in.GetMeasurementMethod(),
-		Operiod: in.GetObservationPeriod(),
-		Unit:    in.GetUnit(),
+	if len(statVars) == 0 {
+		return nil, status.Errorf(
+			codes.InvalidArgument, "Missing required argument: stat_vars")
 	}
 
 	// Read triples for statistical variable.
-	triplesRowList := buildTriplesKey([]string{statVarDcid})
+	triplesRowList := buildTriplesKey(statVars)
 	triples, err := readTriples(ctx, s.btTable, triplesRowList)
 	if err != nil {
 		return nil, err
 	}
-	// Get the StatisticalVariable
-	if triples[statVarDcid] == nil {
-		return nil, status.Errorf(codes.NotFound, "No statistical variable found for %s", statVarDcid)
-	}
-	statVarObject, err := triplesToStatsVar(statVarDcid, triples[statVarDcid])
-	if err != nil {
-		return nil, err
+	// A map from stat_var dcid to stat_var object.
+	// The stat_var object is used to create BigTable cache key
+	statVarObject := map[string]*StatisticalVariable{}
+	for statVar, triplesCache := range triples {
+		if triplesCache != nil {
+			statVarObject[statVar], err = triplesToStatsVar(statVar, triplesCache)
+			if err != nil {
+				return nil, err
+			}
+		}
 	}
 	// Construct BigTable row keys.
-	rowList, keyTokens := buildStatsKey(
-		placeDcids,
-		map[string]*StatisticalVariable{statVarDcid: statVarObject},
-	)
-
-	tmp := map[string]*pb.ObsTimeSeries{}
+	rowList, keyTokens := buildStatsKey(places, statVarObject)
 
+	// Initialize result with place and stat var dcids.
+	result := &pb.GetStatSetSeriesResponse{
+		Data: make(map[string]*pb.SeriesMap),
+	}
+	for _, place := range places {
+		result.Data[place] = &pb.SeriesMap{
+			Data: make(map[string]*pb.Series),
+		}
+		for _, statVar := range statVars {
+			result.Data[place].Data[statVar] = nil
+		}
+	}
 	// Read data from branch in-memory cache first.
 	cacheData := s.memcache.ReadParallel(
 		rowList,
 		convertToObsSeriesPb,
 		tokenFn(keyTokens),
 	)
+
 	for token, data := range cacheData {
-		place := strings.Split(token, "^")[0]
+		parts := strings.Split(token, "^")
+		place := parts[0]
+		statVar := parts[1]
```

Comment:
To confirm, what cache is this that has stat-var ID as second part?

Reply:
The place and statvar here is from the request. They are passed into the memcache parallel read so the read-cache can bind back to the original <place, statvar>